src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java

```diff
@@ -349,6 +350,19 @@ else if(reader.isFlagSet(Constants.LEVEL_01_TOKEN)){
                 lineRepository.addFileSectionStatement(line.getUnNumberedString());
             }
         }
+        if(reader.isFlagSet(Constants.WORKING_STORAGE_SECTION) && 
+        line.containsToken(Constants.EXEC_SQL_TOKEN) && 
+        (line.containsToken(Constants.INCLUDE) || reader.appendNextMeaningfulLineToCurrentLine().containsToken(Constants.INCLUDE))) {
+            lineRepository.addExpandedCopyDB2Statements(reader.readStatementAsOneLine());
+            line.setUnNumberedString("      **");
+        }
+        if(lineRepository.getFileSectionStatements().size() > 0) {
```

Comment:
Please elaborate on these file section changes.

Reply:
Hi Rune, 
I've been working on cobol-check for sometime my company a division is even starting to use it in there development workflow! I just never got around posting those changes here because I can't push stuff on public repos from my work station. So I got to recode the whole thing here on my personal pc.


Basically here are the changes... And I'm not done ;)

***
_In DB2 copybook you have the table and variables as so

           EXEC SQL DECLARE TEXE2 TABLE
           ( FIRST_NAME                  CHAR(10) NOT NULL,
             LAST_NAME                   CHAR(10) NOT NULL,
             WALLET                      INT(10) NOT NULL,
             TMS_CREA                    TIMESTAMP NOT NULL
           ) END-EXEC.
      ******************************************************************
      * COBOL DECLARATION FOR TABLE TEXEM2                             *
      ******************************************************************
       01  TEXE2.
           10 FIRST-NAME           PIC X(10).
           10 LAST-NAME            PIC X(10).
           10 WALLET               PIC 9(8)V99 COMP-3.
           10 TMS-CREA             PIC X(26)._
***

I made sure that only the variables are kept

       _01  TEXE2.
           10 FIRST-NAME           PIC X(10).
           10 LAST-NAME            PIC X(10).
           10 WALLET               PIC 9(8)V99 COMP-3.
           10 TMS-CREA             PIC X(26)._

When the include is found within the program

EXEC SQL INCLUDE COPYNAME END-EXEC
           
I expand the copy book and change the line 

EXEC SQL INCLUDE COPYNAME END-EXEC for a Cobol comment ... ** so the sql statement is ignored.


Also we found out that when ever you expand a copybook the variable that are numeric we always ALPHA so I fixed that
here. When a copybook is expanded le variables are checked for numerics.

            for(int i=0; i < lineRepository.getFileSectionStatements().size(); i++) {
                StringTokenizerExtractor stringTokenizerExtractor = new StringTokenizerExtractor();
                CobolLine cobolLine = new CobolLine(lineRepository.getFileSectionStatements().get(i), stringTokenizerExtractor);
                updateNumericFields(cobolLine);
            }
        }